src/store/actions.js

```diff
@@ -91,44 +92,117 @@ const loadCurrentModel = async ({ commit, state }, payload) => {
+  const { deviceId, productId } = payload
+  commit(types.SET_LOADING, false)
+  commit(types.CLEAR_POLICY_LOADING_INFO)
+
+  const policyLoadingInfo = {
+    deviceId: deviceId
+  }
+
+  commit(types.SET_POLICY_LOADING_INFO, policyLoadingInfo)
 
   let response = null
 
+  // STEP 1: getting task ID
   try {
-    response = await axios.post('insurance/policy-draft/' + imei)
-  } catch (e) {
-    commit(types.LOADING, { loading: false })
+    response = await customAxios.get('insurance/policy/android/pair/' + deviceId)
+  } catch (error) {
+    var newPolicyLoadingInfo = Object.assign({}, policyLoadingInfo)
+
+    if (error.response.status === 404) {
+      newPolicyLoadingInfo.notFound = true
+    } else if (error.response.status === 503 || error.response.status === 500) {
+      newPolicyLoadingInfo.serverError = true
+    }
+
+    commit(types.SET_POLICY_LOADING_INFO, newPolicyLoadingInfo)
+    return
   }
 
-  if (response && response.data.policy) {
-    commit(types.LOAD_CURRENT_POLICY, response.data)
-    commit(types.LOADING, { loading: false })
+  commit(types.CLEAR_POLICY_LOADING_INFO)
+  policyLoadingInfo.taskId = response.data.taskId
+  commit(types.SET_POLICY_LOADING_INFO, policyLoadingInfo)
+
+  // STEP 2: gettings task
+  let waitMultiplier = 1
+  while (!response.data.policyId && !response.data.validationResultCode && state.policyLoadingInfo.taskId) {
+    response = await axios.post('insurance/policy/android', {
+      TaskId: policyLoadingInfo.taskId,
+      ProductId: productId
+    })
+
+    await sleep(waitMultiplier * 250)
+    waitMultiplier++
+  }
+
+  commit(types.CLEAR_POLICY_LOADING_INFO)
+
+  if (response.data.validationResultCode) {
+    policyLoadingInfo.validationResultCode = response.data.validationResultCode
   } else {
-    commit(types.CLEAR_CURRENT_POLICY)
-    commit(types.LOADING, { loading: false })
+    policyLoadingInfo.policyId = response.data.policyId
   }
+
+  commit(types.SET_POLICY_LOADING_INFO, policyLoadingInfo)
+}
+
+const getPolicy = async ({ commit }, policyId) => {
+  commit(types.SET_LOADING, true)
+  commit(types.CLEAR_CURRENT_POLICY)
+  const response = await axios.get('insurance/policy/android/' + policyId)
+
+  if (response && response.data) {
+    commit(types.SET_CURRENT_POLICY, response.data)
+  }
+
+  commit(types.SET_LOADING, false)
+}
+
+const sendPolicyPayment = async ({ commit, state }) => {
+  const web3 = state.userWeb3.web3Instance()
+  const productAddress = '0x146392ae1d5f8894104a531f41ea62e5efa9bbc3'
+
+  const TokenInstance = new web3.eth.Contract(process.env.CONTRACT_INFO.ABI, process.env.CONTRACT_INFO.ADDRESS)
+  const paymentValue = web3.utils.toWei(state.currentPolicy.premium.toString())
+  const policyIdBytes = web3.utils.fromAscii(state.currentPolicy.id)
+
+  TokenInstance.methods
+    .approveAndCall(productAddress, paymentValue, policyIdBytes)
+    .send({ gas: 100, from: state.userWeb3.coinbase })
```

Comment:
gas: 100
not sure, gas maybe configurable too? (just to avoid "redeploying" every time when eth network is crowded and we want to "give" more gas? Just a suggestion.

Reply:
✅ Set to calculated gas limit